### DISCUSSION TITLE
Welp, I'm stuck with my MIDI circuit, any help?

### DISCUSSION THREAD

**Discussion by AndyLAndrew** [probonopd/MiniDexed#296] [Q&A] [answered]
Edit Update (full edit in answer below): Be sure to check your connections/wiring! Also highly recommend checking out warpeggio's advice/finding to use a 56k resistor in place of the 220k. Will play very well with sequencers after making that change. Maybe for future users without any circuitry knowledge, a mockup similar to what diyelectromusic linked below could be provided?

Well I'm back again! I decided to go ahead and build that MIDI IN Circuit based on the posted design in the wiki, but I think I did something wrong? Minidexed will boot but won't show any notification of the port being recognized (I'm not sure if it's supposed to mention it on boot?), but I checked each component on the breadboard with a multi-meter so nothing seems defective (other than my design). I honestly have no idea where to go from here as I have a very elementary understanding of circuits so any help would be welcome!

![IMG_8010](https://user-images.githubusercontent.com/81981636/176358570-407a9435-1252-4305-92fa-7b46f00537b0.jpg)
![IMG_8009](https://user-images.githubusercontent.com/81981636/176358587-18000138-8c1c-4ea8-97f5-bfdcdc6578cd.jpg)
![IMG_8011](https://user-images.githubusercontent.com/81981636/176358599-acbaabaa-2da1-4da7-b6ab-198e8e5c1432.jpg)

**Comment by craigyjp** [comment]
Is it just me or is that 3.3k resistor just connected between ground and 3.3v?

**Reply by AndyLAndrew** [reply]
No it's definitely a confusing perspective! Here's a couple more photos with better angles


![IMG_8205](https://user-images.githubusercontent.com/81981636/176362222-5276f694-42d4-4fe5-bf11-420dca83693b.jpg)
![IMG_8206](https://user-images.githubusercontent.com/81981636/176362239-14db7bae-a3b3-4c3e-a5b6-5e607dad7909.jpg)


Again though I'm really new to circuits so maybe I'm not seeing it!

**Reply by craigyjp** [reply]
Ok, a few things

Swap the midi cables around just be clear you don't have them on the wrong pins, swap them back if no different.

Remove the 220k resistor, just again as a test. 

Are you sure you are transmitting on channel 1 and not something out of the range of the unit. You can set TG1 to Omni mode and save the performance to be certain it's receiving on any channel. 

Add an LED to the output of the midi interface. A simple inverter like a 74HC14 should be enough with a 220R resistor and LED to indicate a good midi signal.

**Reply by craigyjp** [reply]
Also the schematics I have seen show a 1k for the 220k resistor.

https://www.google.com/search?q=raspberry+pi+midi+interface+schematic+&tbm=isch&ved=2ahUKEwjsi-bagNL4AhUpmYsKHSP7CSQQ2-cCegQIABAC&oq=raspberry+pi+midi+interface+schematic+&gs_lcp=ChJtb2JpbGUtZ3dzLXdpei1pbWcQAzoECCMQJzoECAAQHjoECB4QClDXCViSOGCBPWgAcAB4AIABnAGIAYYHkgEDMS43mAEAoAEBwAEB&sclient=mobile-gws-wiz-img&ei=ae27YqyzCKmyrgSj9qegAg&bih=750&biw=412&client=ms-android-samsung-ss&prmd=isvn#imgrc=gE1UsthSf5ByPM

**Reply by AndyLAndrew** [reply]
I'll try most of this out and get back on the results later today thank you! I don't have easy access to an inverter or 1k resistor so that will have to be later this week.

**Reply by AndyLAndrew** [reply]
Well, no dice. Tried all your suggestions but seems to not have any effect. Really hope someone who's replicated this circuit could chime in. Kind of bummed but considering this was the suggested design on the wiki so it has to work, I'm probably just missing something simple. Thanks again for your help
![IMG_8380](https://user-images.githubusercontent.com/81981636/176988922-fef966db-3147-4dc8-bd72-4a7ed053cfc4.jpg)

**Reply by craigyjp** [reply]
Well I have to admit to not ever having much luck with 6n138 optos myself in the past and I've pretty much stuck with 4n35 optos in all my circuits except for earlier interfaces where I used 6n137 chips.

I did post my 4n35 circuit here with additional LED indication and MIDI out and thru options but I think most people feel the 6n138 is the way to go.

**Reply by craigyjp** [reply]
Here is my 4n35 interface....

![Screenshot_20220702-101550_Chrome.jpg](https://user-images.githubusercontent.com/51026305/176990827-2f12c357-8521-4f1f-8f29-b5deb0a5b889.jpg)

**Comment by miotislucifugis** [comment]
going by the photo of the midi jack, (to me) it looks like  you have the center pin connected (orange wire) .   it should be pins #4 and #5.

**Reply by miotislucifugis** [reply]
the center pin can be left unconnected.

**Reply by craigyjp** [reply]
I think again that is an optical illusion, you can just make out the center pin unconnected.

**Reply by AndyLAndrew** [reply]
Sorry! Yes it's another optical illusion, equally awful at circuit photography as I am at circuits, here's a better angle of the MIDI jack
![IMG_8217](https://user-images.githubusercontent.com/81981636/176470002-cdf9ddd5-e591-4b66-a08b-eef8d8fa942f.jpg)

**Comment by miotislucifugis** [comment]
I just checked my 6n138 circuit ( which works well) .....Try replacing R3 -220k  with 1k, and R2 -3.3K with 220 ohms.   Also, I do not have the 10nF cap.

**Reply by AndyLAndrew** [reply]
oh sweet! I'll try this right now. Is it possible to try R3 with 2.2k ohms (any tolerance)? Otherwise I'll have to go get some 1ks or series/parallel what I have already.

**Comment by miotislucifugis** [comment, marked as answer]
try it

**Reply by AndyLAndrew** [reply]
hm no dice (both with parallel and single 2.2k) but I'm guessing there's something else wrong with my circuit. Do you mind sharing a photo of your circuit and/or the schematic? Thank you so much for your help btw

**Reply by miotislucifugis** [reply]
all my current midi  circuits are built onto either homemade pcb or prototyping  board so they may not be much help.   Schematic-wise it is the same as yours, just with the changes i mentioned above.  Not really sure why, but pretty much every schematic ive ever seen for this circuit has a different recipe of resistor values-  they are just determinig the signal levels and response time... even if they are way off it usually just works, but with some missed messages/dropped notes.

Just a thought but check that you are in fact connected to the proper gpio pin on the pi... and double check that you have enabled serial midi in inidexed.ini.   Can you measure the voltage/ continuity and make sure that your circuit is connected to everything it should be?    from my experience, most of those hook up wire use the crappiest, thinnest wire that breaks fairly easily.   very frustrating for troubleshooting.

**Reply by miotislucifugis** [reply]
... also worn breadboard sockets

**Reply by diyelectromusic** [reply]
I don't know if you've solved this one or not, but in case it is of use, everything I've found out about 3V3 compatible MIDI IN circuits, I've written up here: https://diyelectromusic.wordpress.com/2021/02/15/midi-in-for-3-3v-microcontrollers/

Good luck!

Kevin

**Reply by AndyLAndrew** [reply]
- The wires do seem very fragile but I do have multiple extras so I'll swap them all and retest (note all the other components added use the same wire and those work fine but it's worth a shot!).
- Wait, is there an enable midi section in the ini? I checked mine which is using the latest release and there wasn't anything other than Thru or sysex to enable. The wiki also doesn't seem to have anything related to enabling MIDI IN so I'm a wee bit confused haha. I really hope this is the case cause that would be an easy fix.
- I'll remeasure after I swap the components and wiring! Gonna be a bit due to a currently busy schedule
- Breadboard is brand new but I'll shift it to the bottom for testing
- diyelectromusic, this is so helpful thank you! I'll try putting this together and testing it as well.

Thank you so much everyone for all of your help, updates will come soon later this week!


Edit Update (referenced above): Turns out the 3.3v cable was faulty and the 220k resistor was too high for what I was testing it with (MPC Sequencer). Once the 3.3v was fixed, it was working but cutting out. Tested with a MIDI keyboard and appears to work so I followed advice in another discussion thread to try a 56k resistor, worked perfectly! Had a huge bundle of resistors in parallel for testing but it worked!

**Comment by miotislucifugis** [comment]
Did you solve it?   What was the problem?

**Reply by AndyLAndrew** [reply]
Yes! I replied at the second-to-last chain and a short note at the top! Thank you so much for your help! Now to use my newfound circuitry knowledge to build a quick mixer
